src/brevitas/core/scaling/standalone.py

```diff
@@ -412,12 +417,14 @@ def forward(self, stats_input: Tensor, threshold: Optional[Tensor] = None) -> Te
         else:
             if self.counter <= self.collect_stats_steps:
                 out = self.buffer
+                # No clamping is necessary since statistics are already clamped in training_forward
                 out = self.restrict_scaling_pre(out)
             else:
                 out = self.value
             threshold = self.restrict_threshold(self.restrict_threshold_pre(threshold))
-            out = self.clamp_scaling(self.restrict_scaling(out))
```

Comment:
I believe `self.clamp_scaling` is not necessary here because we're already in log domain, and `self.restrict_scaling` is responsible only for rounding and potential conversion from log to real domain (i.e., exponential).

Applying clamping before restrict_val means we can't get to negative values, thus scaling values between 0 and 1.
@nickfraser

Reply:
Sounds right to me!